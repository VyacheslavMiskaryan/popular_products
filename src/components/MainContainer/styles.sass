@import '../../colors'
@import '../../global'

.main-container
    @extend %flex-container
    align-items: center

.main-title
    font-weight: bold
    font-size: 30px

.products-container
    @extend %flex-container
    gap: 15px

.line
    margin-top: -20px
    margin-bottom: 20px
    height: 2px
    width: 100%
    background-color: $signature-color