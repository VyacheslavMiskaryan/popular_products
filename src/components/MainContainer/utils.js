const sortingRule = (field) => (a, b) => (a[field] > b[field] ? -1 : 1);

export default sortingRule;
