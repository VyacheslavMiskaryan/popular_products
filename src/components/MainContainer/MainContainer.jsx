import React from 'react';
import { useSelector } from 'react-redux';

import Product from '../Product';

import sortingRule from './utils';
import './styles.sass';

const MainContainer = () => {
  const { allProducts } = useSelector((state) => state.products);

  return (
    <div className="main-container">
      <p className="main-title">Popular Products</p>
      <div className="line" />
      <div className="products-container">
        {allProducts.sort(sortingRule('votes')).map((item, index) => (
          <Product
            key={item.id}
            title={item.title}
            description={item.description}
            url={item.url}
            votes={item.votes}
            itemIndex={index}
            submitterAvatarUrl={item.submitterAvatarUrl}
            productImageUrl={item.productImageUrl}
          />
        ))}
      </div>
    </div>
  );
};

export default MainContainer;
