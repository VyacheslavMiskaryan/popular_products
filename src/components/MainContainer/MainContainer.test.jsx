import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import MainContainer from './MainContainer';

import store from '../../redux/store';

test('MainContainer renders correctly', () => {
  const component = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <MainContainer />
      </BrowserRouter>
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
