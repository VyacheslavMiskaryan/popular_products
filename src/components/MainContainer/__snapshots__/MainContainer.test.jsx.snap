// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer renders correctly 1`] = `
<div
  className="main-container"
>
  <p
    className="main-title"
  >
    Popular Products
  </p>
  <div
    className="line"
  />
  <div
    className="products-container"
  >
    <div
      className="grid-container"
    >
      <div
        className="image-container"
      >
        <img
          alt="productAvatar"
          src="assets/images/productImages/image-yellow.png"
        />
      </div>
      <div
        className="votes-container"
      >
        <button
          className="MuiButtonBase-root makeStyles-button-1"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 14l5-5 5 5z"
            />
          </svg>
        </button>
        55
      </div>
      <div
        className="description-container"
      >
        <a
          className="title"
          href="/"
          onClick={[Function]}
        >
          <span>
            Haught or Naught
          </span>
        </a>
        <span>
          High-minded or absent-minded? You decide
        </span>
      </div>
      <div
        className="author-signature-container"
      >
        <span>
          Submitted by:
        </span>
         
        <img
          alt="avatar"
          src="assets/images/avatarImages/sarah.png"
        />
      </div>
    </div>
    <div
      className="grid-container"
    >
      <div
        className="image-container"
      >
        <img
          alt="productAvatar"
          src="assets/images/productImages/image-aqua.png"
        />
      </div>
      <div
        className="votes-container"
      >
        <button
          className="MuiButtonBase-root makeStyles-button-1"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 14l5-5 5 5z"
            />
          </svg>
        </button>
        44
      </div>
      <div
        className="description-container"
      >
        <a
          className="title"
          href="/"
          onClick={[Function]}
        >
          <span>
            Yellow Pail
          </span>
        </a>
        <span>
          On-demand sand castle construction expertise
        </span>
      </div>
      <div
        className="author-signature-container"
      >
        <span>
          Submitted by:
        </span>
         
        <img
          alt="avatar"
          src="assets/images/avatarImages/daniel.png"
        />
      </div>
    </div>
    <div
      className="grid-container"
    >
      <div
        className="image-container"
      >
        <img
          alt="productAvatar"
          src="assets/images/productImages/image-gray.png"
        />
      </div>
      <div
        className="votes-container"
      >
        <button
          className="MuiButtonBase-root makeStyles-button-1"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 14l5-5 5 5z"
            />
          </svg>
        </button>
        42
      </div>
      <div
        className="description-container"
      >
        <a
          className="title"
          href="/"
          onClick={[Function]}
        >
          <span>
            Tinfoild: Tailored tinfoil hats
          </span>
        </a>
        <span>
          We already have your measurements and shipping address
        </span>
      </div>
      <div
        className="author-signature-container"
      >
        <span>
          Submitted by:
        </span>
         
        <img
          alt="avatar"
          src="assets/images/avatarImages/vincent.png"
        />
      </div>
    </div>
    <div
      className="grid-container"
    >
      <div
        className="image-container"
      >
        <img
          alt="productAvatar"
          src="assets/images/productImages/image-red.png"
        />
      </div>
      <div
        className="votes-container"
      >
        <button
          className="MuiButtonBase-root makeStyles-button-1"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 14l5-5 5 5z"
            />
          </svg>
        </button>
        23
      </div>
      <div
        className="description-container"
      >
        <a
          className="title"
          href="/"
          onClick={[Function]}
        >
          <span>
            Supermajority: The Fantasy Congress League
          </span>
        </a>
        <span>
          Earn points when your favorite politicians pass legislation
        </span>
      </div>
      <div
        className="author-signature-container"
      >
        <span>
          Submitted by:
        </span>
         
        <img
          alt="avatar"
          src="assets/images/avatarImages/ernest.png"
        />
      </div>
    </div>
  </div>
</div>
`;
